services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/index.tsx

```diff
@@ -0,0 +1,287 @@
+      name: activityPackName,
+      flag: activityPackFlag.value,
+      activity_info: activityPackInfo,
+      time: activityPackTime.value,
+      unit_template_category_id: activityPackType.value,
+      activity_ids: activityPackActivities.map(activity => activity.id),
+      invalidParam: true
+    };
+    if(id) {
+      updateUnitTemplate(unitTemplateObject, id).then(response => {
+        if(response.success) {
+          setModalMessage(SUCCESS_MESSAGE);
+          setShowModal(true);
+        } else {
+          const { error } = response;
+          setModalMessage(error);
+          setShowModal(true);
+        }
+      })
+    } else {
+      createUnitTemplate(unitTemplateObject).then(response => {
+        if(response.success) {
+          setModalMessage(SUCCESS_MESSAGE);
+          setShowModal(true);
+        } else {
+          const { error } = response;
+          setModalMessage(error);
+          setShowModal(true);
+        }
+      })
+    }
+  }
+
+  function handleCloseSubmissionModal() {
+    setShowModal(false);
+  }
+
+  function renderActivityPackPreviewLink() {
+    if(!id) { return };
+    const url = `${process.env.DEFAULT_URL}/assign/featured-activity-packs/${id}`;
+    return(
+      <div className="padded-element">
+        <a className="data-link" href={url} rel="noopener noreferrer" target="_blank">Preview in Featured Activity Pack page</a>
+      </div>
+    );
+  }
+
+  function renderReadabilitySection() {
+    return (
+      <section className="readability-section padded-element">
+        <h3>Readability:</h3>
+        {!readability && <p>{NOT_APPLICABLE}</p>}
+        {readability && <p>{readability}</p>}
+      </section>
+    )
+  }
+
+  function renderDiagnosticsSection() {
+    return (
+      <section className="diagnostics-section padded-element">
+        <h3>Diagnostics:</h3>
+        <section className="diagnostics">
+          {!diagnostic_names && <p>{NOT_APPLICABLE}</p>}
+          {diagnostic_names && diagnostic_names.map((diagnostic) => {
+            return <p>{diagnostic}</p>;
+          })}
+        </section>
+      </section>
+    );
+  }
+
+  function renderPdfUploadSection() {
+    return (
+      <div className="pdf-upload-container padded-element">
+        <label>Click the square below or drag a file into it to upload a file:</label>
+        <Dropzone onDrop={handleFileDrop} />
+        <p>Here is the link to your uploaded file:</p>
+        {uploadedFileLink && <a href={uploadedFileLink} rel="noopener noreferrer" target="_blank">{uploadedFileLink}</a>}
+      </div>
+    )
+  }
+
+  function renderSubmissionModal() {
+    return(
+      <Modal className="rule-view-form-modal">
+        <section className="submission-modal-contents">
+          <p>{modalMessage}</p>
+          <button className="quill-button primary contained fun focus-on-light" onClick={handleCloseSubmissionModal}>Close</button>
+        </section>
+      </Modal>
+    )
+  }
```

Comment:
I think this accomplishes the goal, but it seems like a snackbar would have accomplished the same purpose and been a little more in line with the design language we use elsewhere.

Reply:
good suggestion! I'll update to use a snackbar